### ISSUE TITLE
patchState new value does not reflect in Redux DevTools

### ISSUE THREAD

**Issue by kuncevic** [ngxs/store#329]
### Versions
```
* ngxs: 3.0.0-rc.1 and 3.0.0-dev.master-2df48fe
* @angular/core: 6.0.0-rc.5
```

### Repro steps

Have `loading: false` in the state.
Patch the state  from an action `sc.patchState({ loading: true });`

### Observed behavior

 The Redux DevTools still show `loading: false` 

### Desired behavior

 Redux DevTools shold show `loading: true`

**Comment by markwhitfeld** [comment]
This may be related in part to issue #311.
Is this behavior only evident when using the patch method, or does it happen if you use setState too?

**Comment by kuncevic** [comment]
If I `@select` the store I am getting the new updated value, so there is only dev-tools does not updating/syncing with a new value in case `patchState`. Cant see any issues with `setState` atm.

**Comment by kuncevic** [comment]
Just updated to rc.4 the issue still there

**Comment by amcdnl** [comment]
I don’t know how It’s possible this is still not working. And just to confirm you said it worked fine with set state?

**Comment by kuncevic** [comment]
All good now

**Comment by garthmason** [comment]
I confirm I am not seeing any issues using `patchState` with the following versions:

`"@angular/core": "6.0.0"`
and 
`"@ngxs/devtools-plugin": "^3.0.0-rc.4"`
` "@ngxs/logger-plugin": "^3.0.0-rc.4"`
` "@ngxs/store": "^3.0.0-rc.4",`

**Comment by GreenMonkeyBoy** [comment]
Looks like it happens if you use patchState() or setState() in the callback methods of a http request.

Demo 1: https://stackblitz.com/edit/ngxs-devtools-not-reflecting-state-subscribe
In this demo, I use subscribe() in the action handler (not sure if it's recommended though), then I use setState() but the devtools is always "one state behind" the state in the real store.

Demo 2: https://stackblitz.com/edit/ngxs-devtools-not-reflecting-state-tap
In this demo, I use tap() like in the [Authentication recipe](https://ngxs.gitbook.io/ngxs/recipes/authentication) from the doc. The devtools is sync on a succeed request, but nothing happens on a failed request.

Hope it will help you.

**Comment by piernik** [comment]
I have the same problem. Any news?

**Comment by arturovt** [comment]
@kuncevic Is it still reproducable? Because I cannot reproduce it :(

**Comment by arturovt** [comment]
@GreenMonkeyBoy @garthmason @kuncevic gentlemen, would it be possible to test with the latest `@dev` version and confirm if the issue still exists? I've just merged the above PR that should fix this issue.

**Comment by arturovt** [comment]
@GreenMonkeyBoy 

I've just installed the latest `@dev` version in your provided example and I can confirm it works:

![image](https://user-images.githubusercontent.com/7337691/63552070-6b943f80-c53f-11e9-9c76-9b94e030568a.png)

**Comment by splincode** [comment]
I can also confirm that this works.

**Comment by markwhitfeld** [comment]
Please could you confirm that this is fixed with the v3.5.1 release?

**Comment by cedvdb** [comment]
It doesn't work on version v3.5.1 for me.

````
  @Action(GetAllRelationships)
  getAllRelationships(ctx: StateContext<RelationshipStateModel>, { user }: GetAllRelationships) {
    return this.afStore.collection<Relationship>(Col.RELATIONSHIP, ref => ref.where('users', 'array-contains', user.id))
      .valueChanges()
      .pipe(
        tap(relations => ctx.setState({ pending: false, data: relations })),
      );
  }
```
